Android/app/src/main/java/com/example/android/activities/MenuActivity.java

```diff
@@ -0,0 +1,135 @@
+package com.example.android.activities;
+import com.example.android.models.MenuItem;
+import com.example.android.adapters.MenuItemAdapter;
+import com.example.android.R;
+import com.example.android.api.ApiService;
+import com.example.android.api.RetrofitClient;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import java.util.ArrayList;
+import java.util.List;
+import android.os.Bundle;
+import android.widget.Button;
+import androidx.activity.OnBackPressedCallback;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+import android.util.Log;
+import android.view.Menu;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+public class MenuActivity extends AppCompatActivity {
+    private RecyclerView recyclerView;
+    private MenuItemAdapter menuItemAdapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_menu);
+
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+            getSupportActionBar().setDisplayShowHomeEnabled(true);
+            getSupportActionBar().setTitle("");
+        }
+
+        OnBackPressedCallback callback = new OnBackPressedCallback(true) {
+            @Override
+            public void handleOnBackPressed() {
+                finish();
+            }
+        };
+        getOnBackPressedDispatcher().addCallback(this, callback);
+
+        Button buttonRamen = findViewById(R.id.buttonRamen);
+        Button buttonUdon = findViewById(R.id.buttonUdon);
+        Button buttonRice = findViewById(R.id.buttonRice);
+
+        recyclerView = findViewById(R.id.recyclerViewMenuItems);
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+        menuItemAdapter = new MenuItemAdapter(null, "");
+        recyclerView.setAdapter(menuItemAdapter);
+
+        buttonRamen.setOnClickListener(v -> {
+            updateButtonSelection(buttonRamen, buttonUdon, buttonRice);
+            fetchMenuItems("RAMEN");
+        });
+
+        buttonUdon.setOnClickListener(v -> {
+            updateButtonSelection(buttonUdon, buttonRamen, buttonRice);
+            fetchMenuItems("UDON_NOODLES");
+        });
+
+        buttonRice.setOnClickListener(v -> {
+            updateButtonSelection(buttonRice, buttonRamen, buttonUdon);
+            fetchMenuItems("RICE_NOODLES");
+        });
+        buttonRamen.performClick();
+    }
+
+    private void updateButtonSelection(Button selectedButton, Button... otherButtons) {
+        selectedButton.setBackgroundResource(R.drawable.button_background_selected);
+        selectedButton.setTypeface(null, android.graphics.Typeface.BOLD);
```

Comment:
this also create universal method because ypu are doing the same thing with one vary param